src/Raven.Server/Documents/Document.cs

```diff
@@ -51,6 +51,30 @@ public unsafe ulong DataHash
         public bool TryGetMetadata(out BlittableJsonReaderObject metadata) =>
             Data.TryGet(Constants.Documents.Metadata.Key, out metadata);
 
+        
+        
+        public void EnsureDocumentId()
+        {
+            if (_metadataEnsured)
+                return;
+
+            _metadataEnsured = true;
```

Comment:
is this correct? is there a chance that we will want full metadata afterwards?

Reply:
They both modify the metadata, so they cannot both be called

Reply:
What if I will change the test your added to assert for `@last-modified` ?

Reply:
It will be null in both of them.
We aren't adding metadata in scripts select, so I matched the behavior